### ISSUE TITLE
First `yarn devserver` on new KOLIBRI_HOME fails first time only

### ISSUE THREAD

**Issue by nucleogenesis** [learningequality/kolibri#7302]
### Observed behavior

I've seen this before and thought it was to do with me getting crazy with changing my KOLIBRI_HOME and branches because I used to just run all of my pip and yarn installs then retry starting the dev server - but now that I've seen it a few times I figure I should make an issue - especially after it just worked without me doing all of that reinstalling of dependencies.

1. Set your KOLIBRI_HOME to a non-existent folder
2. `yarn devserver`
3. See that it fails:

```
Traceback (most recent call last):
  File "/home/jacob/.pyenv/versions/3.7.7/bin/kolibri", line 11, in <module>
    load_entry_point('kolibri', 'console_scripts', 'kolibri')()
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/site-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2843, in load_entry_point
    return ep.load()
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2434, in load
    return self.resolve()
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2440, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/jacob/Code/LearningEquality/kolibri/kolibri/utils/cli.py", line 29, in <module>
    from . import sanity_checks
  File "/home/jacob/Code/LearningEquality/kolibri/kolibri/utils/sanity_checks.py", line 10, in <module>
    from .conf import KOLIBRI_HOME
  File "/home/jacob/Code/LearningEquality/kolibri/kolibri/utils/conf.py", line 42, in <module>
    os.mkdir(KOLIBRI_HOME)
FileExistsError: [Errno 17] File exists: '/media/hdd/kolibris/14-informal'
```

4. `yarn devserver` - now it works as expected.

Looks like the new home folder is being made twice without something catching the error when it tries and fails after it first finds that the folder doesn't exist yet?

### Expected behavior

I make a new KOLIBRI_HOME and can start the devserver on the first try.

### User-facing consequences

Confusing for developers and contributors.

### Context

0.14-beta9
Debian 10
Chrome

**Comment by jonboiser** [comment]
This is a little different from how I've experienced this problem. Usually it's because of a database problem, which I could fix by running `kolibri manage migrate` immediately after deleting the `KOLIBRI_HOME` folder.

**Comment by rtibbles** [comment]
I am fairly sure that this would be fixed by #7254 - where for the devserver we use the production kolibri server running in autoreload mode. We already have safeguards against a non-existent Kolibri home in our production environment, so this would give a parallel experience for devs.

**Comment by jonboiser** [comment]
I experience this issue a little differently than Jacob, where basically the `devserver` script fails on the first run with this error

```
cherrypy.process.wspbus.ChannelFailures: OperationalError('no such table: morango_databaseidmodel',)
```

My hypothesis is that the migrations are not being executed (I see no logs from migrations), so the devserver is being run with a non-existent or incompletely structured DB. So I'll just aim to fix that DB issue.

**Comment by rtibbles** [comment]
This probably happens because we currently use `--skip-update` when running the devserver, so migrations don't run.

**Comment by jonboiser** [comment]
How should we handle this? Does using `--skip-update` flag assume something about the system, like that `kolibri start` was run at least once, so everything is already setup?

I think what would fix my situation is that, if KOLIBRI_HOME does not exist, then I would overwrite `skip_update = False` in `cli::initialize`. 

This will cause the code to into this branch and run the migrations.

https://github.com/learningequality/kolibri/blob/781cadfce5c56911914e613ae26b3aef36bb3333/kolibri/utils/cli.py#L357

I think I also need to make sure this code runs correctly when running from scratch:

https://github.com/learningequality/kolibri/blob/781cadfce5c56911914e613ae26b3aef36bb3333/kolibri/utils/cli.py#L329-L332

Since `updated` seems to get set to `True` in my testing. This way, we avoid running the other upgrade routines unnecessarily.

That is, if the KOLIBRI_HOME is empty (and hence no `.kolibri_version` file), `updated` is set to False.

**Comment by rtibbles** [comment]
Yeah, running `--skip-update` assumes that someone else is taking care of initialization, so I'd rather keep its strong behaviour - it's useful for preventing accidentally running a migration when someone else is already taking care of it. I think given that we have removed collectstatic from our upgrade routines, it's actually safe/not very time consuming to just remove the `--skip-update` flag from development now.

**Comment by jonboiser** [comment]
Ok, that's pretty simple and seems to work the same as what I was thinking of doing.

**Comment by jonboiser** [comment]
Fixed in #7938, specifically for the non-existent KOLIBRI_HOME folder case.
